route/route_test.go

```diff
@@ -234,3 +235,63 @@ func TestUnmarshal(t *testing.T) {
 		t.Error("Expecting", now, "Received", b)
 	}
 }
+
+func TestGetAPIKeyAndDatasetFromMetadataCaseInsensitive(t *testing.T) {
```

Comment:
I'd like to see some tests covering the parsing of the sample rate from the OTLP request - especially since we're doing some type assertions.

Reply:
Added testing for getting sampleRatee from attr. Includes missing and different value types 👍